ui/raidboss/emulator/overrides/RaidEmulatorOverlayApiHook.js

```diff
@@ -33,11 +33,29 @@ export default class RaidEmulatorOverlayApiHook {
           // nextSignificantState is a bit inefficient but given that this isn't run every tick
           // we can afford to be a bit inefficient for readability's sake
           const combatantState = combatant.nextSignificantState(timestamp).toPluginState();
-          if (msg.ids && msg.ids.includes(parseInt(id, 16)))
-            combatants.push(combatantState);
-          else if (msg.names && msg.names.includes(tracker.combatants[id].name))
+          combatantState.ID = combatant.id;
+          combatantState.Name = combatant.name;
+          combatantState.Level = combatant.level;
+          combatantState.Job = combatant.job;
```

Comment:
I think either this or Trim21's version is fine.  If there's a type issue, we can handle it when this file gets converted to TypeScript and it's clearer what the tradeoffs are.

Am I correct that this is the fix? There's a bunch of other changes going on, but this is fundamentally the fix?

Reply:
Yes, this fixes Trim21's original issue. The other changes include a fix for other triggers as discussed on the comments below this one, and fixes to the logic to more closely match the OverlayPlugin implementation.